src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+        # Note: for polar view, no PIXEL_SIZE nor POS
+
+        new_md = {}
+        if model.MD_ACQ_DATE in md:
+            new_md[model.MD_ACQ_DATE] = md[model.MD_ACQ_DATE]
+        if model.MD_POL_MODE in md:
+            new_md[model.MD_POL_MODE] = md[model.MD_POL_MODE]
+
+        return new_md
+
+    def _onBackground(self, data):
+        """
+        Called after the background has changed.
+        :param data: (list) List of data arrays.
+        """
+        self._shouldUpdateImage()
+
+    def _onPoint(self, pos):
+        """
+        Called when a new ebeam position has been selected.
+        :param pos: () TODO
+        """
+        self._shouldUpdateImage()
+
+    def _getBackground(self, pol_mode):
+        """
+        Get the background image from the .background VA on the stream.
+        It must match the polarization position.
+        :param pol_mode: (str) The polarization mode, the background is requested for.
+        :return: (DataArray or None) The background image corresponding to the requested polarization
+                position or None, if no matching background can be found.
+        """
+        bg_data = self.stream.background.value  # list containing DataArrays, DataArray or None
+
+        if bg_data is None:
+            return None
+
+        if isinstance(bg_data, model.DataArray):
+            bg_data = [bg_data]  # convert to list of bg images
+
+        for bg in bg_data:
+            # if no analyzer hardware, set MD_POL_MODE = "pass-through" (MD_POL_NONE)
+            if bg.metadata.get(model.MD_POL_MODE, model.MD_POL_NONE) == pol_mode:
+                # should be only one bg image with the same metadata entry
+                return bg  # DataArray
+
+        # Nothing found e.g. pol_mode = "rhc" but no bg image with "rhc"
+        logging.debug("No background image with polarization mode %s ." % pol_mode)
+        return None
+
+    def _processBackground(self, data, pol_mode):
+        """
+        Process the background on the raw data. Try to get the background image on the stream
+        and substracts it if available. Otherwise process do a simple background processing.
+        :param data: (DataArray) The data that will be background corrected.
+        :param pol_mode: (str) The polarization mode, the background is requested for.
+        :return: (DataArray) The background corrected data.
+        """
+        bg_image = self._getBackground(pol_mode)
+        if bg_image is None:
+            # Simple version: remove the background value
+            data_corr = angleres.ARBackgroundSubtract(data)
+        else:
+            if data.shape != bg_image.shape:
+                # resize
+                bg_image = img.rescale_hq(bg_image, data.shape)
+            data_corr = img.Subtract(data, bg_image)  # metadata from data
+
+        return data_corr
+
+
+class ARRawProjection(ARProjection):
+    """
+    An ARProjection is a typical projection used to show 2D angle resolved images.
```

Comment:
The most important point to explain here is what is the difference between a ARProjection and a ARRawProjection. Why would someone want to create a ARRawProjection?

Reply:
Yes forgot to do that. Might still need some polishing though.